### PR TITLE
[optimize]: update github action

### PR DESCRIPTION
**Pull request by yuluo-yx**

### CODE REVIEW

**Comment by yuluo-yx** [comment]
@zhangshenghang @tom ptal.

File: .github/workflows/doc-build-test.yml [hunk 1]
```diff
@@ -28,103 +28,43 @@ on:
...
         run: |
           sudo npm install -g markdown-link-check@3.8.7
           for file in $(find ./home -name "*.md"); do
-            if ! grep -Fxq "$file" .github/exclude_files.txt; then
-              markdown-link-check -c .github/link_check.json -q "$file"
+            if ! grep -Fxq "$file" ./script/ci/exclude_files.txt; then
+              markdown-link-check -c ./script/ci/link_check.json -q "$file"
             fi
           done
+        # uses: JustinBeckwith/linkinator-action@v6
```
Comment (zhangshenghang) [inline .github/workflows/doc-build-test.yml:57]: Unused ones can be deleted directly

**Review by zhangshenghang — CHANGES_REQUESTED** [review]
I think the checking rules should also be described in the corresponding documents, so that other developers can follow them.

File: .github/workflows/doc-build-test.yml (hunk 1 above)
Comment (yuluo-yx) [reply .github/workflows/doc-build-test.yml]: done!

**Review by zhangshenghang — APPROVED** [review]

**Pull request merged**